### ISSUE TITLE
soup is too slow on raspberry pi

### ISSUE THREAD

**Issue by primaeval** [vonH/plugin.video.iplayerwww#41]
I've been experimenting with my soup version on a Raspberry Pi and it is just too slow. 
I'll see if I can get the optimised parsers to work but I don't think it should go into the main release yet.

**Comment by CaptainTK** [comment]
Too bad and kind of weird as it feels faster than the regexps on my computer.

Which version of Raspberry Pi do you use? If it is not Pi 2, I wouldn't bother. The other versions are simply too slow for anything useful.

**Comment by primaeval** [comment]
I'm using a couple of rpi first generations. Everything is slooowww with them but just good enough for the kids. I'm hoping it might just be downloading more images than before. I'll have to do some profiling to find out.
I'm glad you like it on your computer. I think the progress dialog for long operations like search helps.

**Comment by primaeval** [comment]
That was painful. Everything was taking a few minutes to complete.
I've made a "regex" branch that does the same as the soup branch but with regexes.
Beware of the lookahead assertion to match html tags.
It should be maintainable but pretty fast.
I can use it on the rpi without the pain.

**Comment by CaptainTK** [comment]
Sorry for not replying earlier, this must have slipped my mind.

I admire your enthusiasm to start yet another new development. One of the questions that is open to for is whether to continue with regexps or change to soup. Frankly speaking, I would prefer soupp because it would also allow to parse the playlists much easier that with regexps. However, if there is a serious performance regression, I would stick to regexps. What I don't want to do is maintain both soup and regexps.

Did you ever find out what the problem on the Pi was? Was it the different parsers? Memory bandwidth? Something else?

**Comment by primaeval** [comment]
I did a profile on the rpi soup branch. It took close to an hour or something stupid I think. The html parser does hundreds of regexes on the DOM which is taking the most time. In real time even opening Highlights on a rpi takes a couple of minutes. It is just too long. I really can't recommend soup. There are millions of rpi users so it is not really a niche market. I hate to not recommend my work but I just can't in this case.

My regex branch still works. I just tested it. It condenses all the different search functions down into one and all the highlights searches into another: ScrapeEpisodes and ListHighlights. It should be easier to maintain than the master branch. The searches are smaller so it should be easier to find any future problems. I think this is the way to go.
